src/main/java/com/palantir/gradle/gitversion/GitVersionPlugin.java

```diff
@@ -1,31 +1,32 @@
 package com.palantir.gradle.gitversion;
 
 import groovy.lang.Closure;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import org.eclipse.jgit.api.Git;
 import org.eclipse.jgit.internal.storage.file.FileRepository;
 import org.gradle.api.Action;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
 
-import java.io.File;
-import java.io.IOException;
-
 public class GitVersionPlugin implements Plugin<Project> {
-    public void apply(final Project project) {
+    private final Map<GitVersionArgs, Map<File, VersionDetails>> cachedVersionDetails = new HashMap<>();
```

Comment:
How does this interact with daemons? Is there a way to reset this on every build start, or will the daemon do this automatically?